Remove AMD restrictions on triton hashing

oulgen opened the pull request:
Summary: When we added these functions, AMD's triton checkout was very old, it appears to have caught up. Remove restrictions.

Test Plan: unit tests

Differential Revision: D61351473

nmacchioni reviewed (commented):
LGTM, should be good since we moved to upstream repo

oulgen committed:
Remove AMD restrictions on triton hashing (#133616)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/133616

When we added these functions, AMD's triton checkout was very old, it appears to have caught up. Remove restrictions.

Test Plan: unit tests

Reviewed By: nmacchioni, mxz297

Differential Revision: D61351473
